### ISSUE TITLE
`Request` object behaves differently in fetch

### ISSUE THREAD

**Issue by lennartkerkvliet** [vercel/next.js#66840]
### Link to the code that reproduces this issue

https://github.com/lennartkerkvliet/next-fetch-request-bug

### To Reproduce

When sending a request body using the `Request` object in a Next.js application, the content-length parameter is not included in the request and instead chunked transfer encoding is used. This behavior differs from the expected functionality of the Fetch API, which internally is also constructing a `Request` object.

1. Create a `Request` object with a body.
2. Pass the `Request` object to the `fetch` function.

Example code:
```javascript
const body = JSON.stringify({ username: "test", password: "password" });
const request = new Request("https://www.my-site.com/api/login", { 
    method: "POST", 
    headers: { "Content-Type": "application/json" },
    body: body
});
const response = await fetch(request); // Server receives an empty body
```

### Current vs. Expected behavior

I would expect the parameters in `Request` to be identical to when using the more traditional use of fetch API:

```javascript
const body = JSON.stringify({ username: "test", password: "password" });
const response = await fetch("https://www.my-site.com/api/login", { 
    method: "POST", 
    headers: { "Content-Type": "application/json" },
    body: body
});
```

but instead will return the latter

<img width="1505" alt="Screenshot 2024-06-13 at 18 27 51" src="https://github.com/vercel/next.js/assets/5382580/39492428-d95d-412c-9683-5ddac824a7c9">

### Provide environment information

```bash
Operaring System:
  Arch: arm64
  Version: macOS Sonoma 14.4.1

Binaries:
  Node: v22.2.0
  npm: 10.7.0

Relevant Packages:
  next: 14.2.3
```

### Which area(s) are affected? (Select all that apply)

Not sure, Runtime

### Which stage(s) are affected? (Select all that apply)

next dev (local), Vercel (Deployed), Other (Deployed)

### Additional context

- Using plain Node.js, the above code works as expected.

It seems to be somewhat related to the changes NextJS is making to the `Response` object and in particular the `duplex` option.  The Django server I am using for my own project does not support `transfer-encoding`, and will mark the request as a "Bad request syntax".

**Comment by icyJoseph** [comment]
I can't reproduce with 14.2.4 and node 22.1.0.

I used a request bin: https://public.requestbin.com/r/enhaa8qwmvxi7/2hpuXZ1mqrzZs3xYQrlXRux6jny

And as you can see, content length was always respected, when using:

```js
export default async function Home() {
  const body = JSON.stringify({ information: "test-abc" });
  const url = "https://enhaa8qwmvxi7.x.pipedream.net";
  const options = {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: body,
  };

  // await fetch(url, options);
  await fetch(new Request(url, options));

  return <div>Hi</div>;
}

```

I also tested with 14.2.3 - does it have to be within a server action?

**Comment by icyJoseph** [comment]
Indeed, it has to be using a server action... 

- https://public.requestbin.com/r/enjt3nvh6tmy7/2hpvcwEfXRfc2R5fF9QRerk7fY7

![Image](https://github.com/vercel/next.js/assets/21013447/b87e02b4-076e-4a81-91b0-94aadf10f1b5)

**Comment by MahadAdil** [comment]
Hi there,
Apparently reproducing your code throws the following error:

`Error: Only plain objects, and a few built-ins, can be passed to Client Components from Server Components. Classes or null prototypes are not supported.`

This is because in Next.js in order to protect client-server boundary, Next.js restricts passing Non-serializable value from client to server and since Request() object is non-serializable your code throws an error. Upon refactoring your code within "use server" block, everything works as expected and the server returns status 200 upon POST request.

Working Code:
`
const body = JSON.stringify({ username: "test", password: "password" });

async function brokenAction() {
    "use server"

    // Refactored the Request Object within "use-server" scope
    const request = new Request("https://www.my-site.com/api/login", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: body
    });

    // this will *not* include the body in the request
    await fetch(request)
  }
`

Kindly let me know if this helps or if I missed anything.

**Comment by lennartkerkvliet** [comment]
@icyJoseph You're correct. When rewriting, I forgot to add the context that this only happens in the server environment, which is where [NodeJS Undici](https://github.com/nodejs/undici) is being used. This behavior is internally changed by next and that seems to be causing this issue.

**Comment by lennartkerkvliet** [comment]
@MahadAdil Did you check out the [included sample code](https://github.com/lennartkerkvliet/next-fetch-request-bug)? Route requests (in the App Router) do allow chunked encoding, but you can see that the two server actions which should send the same Request, send different ones. The screenshot of the diff in my description comes from that sample code as well.

**Comment by icyJoseph** [comment]
@lennartkerkvliet not quite, you see, it also works fine, if you make a request as you render, in the server. At least from what I remember testing yesterday.

The issue happens, when the request is made within a server action, not exactly the same thing as above.

**Comment by lennartkerkvliet** [comment]
@icyJoseph you mean it works fine when using SSR but not when using Server Actions? I didn't test it, but it would make sense as the patch applied by next only happens in the server action environment AFAIK.

**Comment by parhammmm** [comment]
I can confirm this happens with the above code in `15.2.4` request body is empty, when request is constructed like @lennartkerkvliet has described

**Comment by vitoras** [comment]
Any updates on this?

_Edit by maintainer bot: Comment was **automatically** minimized because it was considered unhelpful. (If you think this was by mistake, let us know). Please only comment if it adds context to the issue. If you want to express that you have the same problem, use the upvote 👍 on the issue description or subscribe to the issue for updates. Thanks!_
